src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearing/UnlistHearingAction.java

```diff
@@ -120,16 +114,15 @@ public void act() {
         hearing.setStatus(statusConfigService.getStatusConfig(Status.Unlisted));
 
         originalHearingParts = userTransactionDataPreparerService.mapHearingPartsToStrings(objectMapper, hearingParts);
-        hearingParts.stream().forEach(hp -> {
+        hearingParts.forEach(hp -> {
             hp.setSession(null);
-            hp.setSessionId(null);
             hp.setStart(null);
             VersionInfo vi = getVersionInfo(hp);
             hp.setVersion(vi.getVersion());
```

Comment:
The same thing here with optimistic lock, not sure if it will work without detaching. And also there's a high possibility that you need to use hearingPartRepository to make it work.

Reply:
Detached the hearing. Still works without hearingPartRepository